backend/Dockerfile

```diff
@@ -0,0 +1,36 @@
+# Get image for Python
+FROM python:3.8-slim-buster
+
+# Set working directory
+WORKDIR /app/
+
+# pip does not like being run as root, so create a user
+RUN useradd --create-home jina
+
+# Add the data folder locally to container
+COPY ./data /app/data
+
+# Give read permissions for the data folder
+RUN chown jina data
+
+# Switch to user
+USER jina
+
+# Path change needed for huggingface-cli and jina
+ENV PYTHONPATH "${PYTHONPATH}:/home/jina/.local/bin"
+ENV PATH "${PATH}:/home/jina/.local/bin"
```

Comment:
Can you clarify why we need these lines? I didn't need that till now.

Reply:
`pip` does not like being run as `root`, so I made a user inside the container on startup that creates a user called `jina`. In the first case where pip was being run as `root`, dependencies get installed globally under `/usr/bin`, but this is not usually how dependencies are installed for user with lower privileges than `root`. When I installed the dependencies as the `user`, the container and python threw errors that were looking for some dependencies under these two variables in these two locations. A quick solution was just to append to these paths so those go away.

I was initially expecting the user dependencies to be installed under `/bin`, but that was not the case.

Reply:
Ok, thanks for the explanation!

This is a very general problematic, what is the accepted way to deal with it? (If there is one!)
Running as `root` is probably not what we want, so I see the need for a new user. But it should be possible to switch to that user before running `pip` in such a way that there are no `PATH` problems (maybe running `pip --user install ...`?).

Reply:
I tried a few combinations, but I could not find a way where it automatically appended to the $PATH variable.

I think for the most part, this appending to the $PATH variable will not be problematic.

Though, I am not sure why pip does not recognize appending to the path itself.

Do you think this is worth making an issue on the `pip` github repository? I think one way of testing this would be creating a VM with `Vagrant`, `qemu`, or `VirtualBox`, and installing a bare minimum python version and then `pip` to check if this is reproducible or just a problem with `Docker`.

Reply:
Going with the previous process.